### PR TITLE
Add unit tests for configuration system and record types

### PR DESCRIPTION
**Pull request by shantanuparab-tr**
### TL;DR

Added comprehensive unit tests for configuration system and record types to improve test coverage.

### What changed?

Added two new test executables:
- `test_config.cpp` - Tests for `SessionManagerConfig`, `GlobalConfig`, and `ConfigRegistry` including JSON parsing, default values, partial configurations, type safety, and error handling
- `test_record_types.cpp` - Tests for `TeleopJointStateRecord` and `Odometry2DRecord` covering construction, type conversion, default values, and polymorphic behavior

Both test executables are integrated with GoogleTest and CTest for automated test discovery.

### How to test?

Run the new tests using:
```bash
ctest -R "test_config|test_record_types"
```

Or run the executables directly:
```bash
./test_config
./test_record_types
```

### Why make this change?

These tests provide essential coverage for configuration management and data record functionality that wasn't previously tested, ensuring reliability of JSON configuration parsing, type safety in the configuration system, and correct behavior of record types used throughout the SDK.

### CODE REVIEW

**Comment by shantanuparab-tr** [comment]
* **#193** <a href="https://app.graphite.com/github/pr/TrossenRobotics/trossen_sdk/193?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#191** <a href="https://app.graphite.com/github/pr/TrossenRobotics/trossen_sdk/191?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#190** <a href="https://app.graphite.com/github/pr/TrossenRobotics/trossen_sdk/190?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#189** <a href="https://app.graphite.com/github/pr/TrossenRobotics/trossen_sdk/189?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#188** <a href="https://app.graphite.com/github/pr/TrossenRobotics/trossen_sdk/188?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#187** <a href="https://app.graphite.com/github/pr/TrossenRobotics/trossen_sdk/187?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.com/github/pr/TrossenRobotics/trossen_sdk/187?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* **#186** <a href="https://app.graphite.com/github/pr/TrossenRobotics/trossen_sdk/186?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#185** <a href="https://app.graphite.com/github/pr/TrossenRobotics/trossen_sdk/185?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#184** <a href="https://app.graphite.com/github/pr/TrossenRobotics/trossen_sdk/184?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#183** <a href="https://app.graphite.com/github/pr/TrossenRobotics/trossen_sdk/183?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#182** <a href="https://app.graphite.com/github/pr/TrossenRobotics/trossen_sdk/182?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#181** <a href="https://app.graphite.com/github/pr/TrossenRobotics/trossen_sdk/181?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`


<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

**Review by lukeschmitt-tr — APPROVED** [review]

File: tests/test_record_types.cpp [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+}
+
+// Ensure RecordBase fields inherited correctly
+TEST(Odometry2DRecordTest, InheritedFields) {
+  Odometry2DRecord rec;
+  rec.ts = trossen::data::make_timestamp_now();
+  rec.seq = 77;
+  rec.id = "test_odom";
+
+  EXPECT_GT(rec.ts.monotonic.sec, 0);
```
[the remark this answers is not in the document]
Comment (shantanuparab-tr) [reply tests/test_record_types.cpp]: Updated the assertion to handle the edge case where sec == 0 but nsec > 0.

File: tests/test_config.cpp [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+  nlohmann::json config = {
+    {"session_manager", {
+      {"type", "session_manager"},
+      {"max_duration", 15.0},
+      {"backend_type", "null"}
+    }}
+  };
+
+  // NOTE: This modifies the global singleton. Since other test suites
+  // also load configs, we accept that the last load wins.
```
[the remark this answers is not in the document]
Comment (shantanuparab-tr) [reply tests/test_config.cpp]: Updated the comment to clarify the merge/overwrite behavior of load_from_json().

**Comment by shantanuparab-tr** [comment]
### Merge activity

* **Mar 17, 2:26 PM UTC**: A user started a stack merge that includes this pull request via [Graphite](https://app.graphite.com/github/pr/TrossenRobotics/trossen_sdk/187).
* **Mar 17, 2:37 PM UTC**: [Graphite](https://app.graphite.com/github/pr/TrossenRobotics/trossen_sdk/187) rebased this pull request as part of a merge.
* **Mar 17, 2:38 PM UTC**: @shantanuparab-tr merged this pull request with [Graphite](https://app.graphite.com/github/pr/TrossenRobotics/trossen_sdk/187).
